awscli/customizations/s3/utils.py

```diff
@@ -760,6 +760,33 @@ def on_queued(self, future, **kwargs):
         future.meta.provide_transfer_size(self.size)
 
 
+class OnDoneFilteredSubscriber(BaseSubscriber):
```

Comment:
I think a TODO note along with marking it internal would be reasonable for now.

The main confusing part was the fact that it followed the same naming convention `on_<event>` as all the built in subscriber methods but was actually tied to a specific subclass.  If these were named `_on_success`, that would at least be a clue that this is different that the built in subscribers.